client/rest-high-level/src/test/java/org/elasticsearch/client/watcher/GetWatchResponseTests.java

```diff
@@ -0,0 +1,285 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.elasticsearch.client.watcher;
+
+import org.elasticsearch.common.bytes.BytesReference;
+import org.elasticsearch.common.xcontent.DeprecationHandler;
+import org.elasticsearch.common.xcontent.LoggingDeprecationHandler;
+import org.elasticsearch.common.xcontent.NamedXContentRegistry;
+import org.elasticsearch.common.xcontent.ToXContent;
+import org.elasticsearch.common.xcontent.ToXContentObject;
+import org.elasticsearch.common.xcontent.XContent;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentFactory;
+import org.elasticsearch.common.xcontent.XContentParser;
+import org.elasticsearch.common.xcontent.XContentType;
+import org.elasticsearch.test.AbstractXContentTestCase;
+import org.joda.time.DateTime;
+import org.joda.time.DateTimeZone;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.HashMap;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Objects;
+
+import static org.elasticsearch.index.mapper.DateFieldMapper.DEFAULT_DATE_TIME_FORMATTER;
+
+public class GetWatchResponseTests extends AbstractXContentTestCase<GetWatchResponseTests.ResponseWrapper> {
```

Comment:
I think the new standard is to extend `AbstractHlrcStreamableXContentTestCase`. @vladimirdolzhenko introduced this testing option, and had linked to 'put license' as an example: https://github.com/elastic/elasticsearch/pull/34547/files#diff-1ed9d64125ce162773bfc2a4b02f5783

Reply:
for this to work we'd need to implement `toXContent` on the HLRC `GetWatchResponse` and by extension to `WatchStatus`. I can add it but I thought that we've decided not to add this if it's solely used in tests ?

Reply:
This new testing approach wasn't available when we had that discussion (which I think was part of the 'ack watch' PR). The test works by depending on both the HLRC and transport classes. I think it creates random instances of the transport `GetWatchResponse`, calls `toXContent` on them, and then checks that the HLRC `GetWatchResponse` can correctly parse the output.

One slight issue is that the transport `GetWatchResponse` doesn't currently implement `ToXContent`. But I think this wouldn't be too hard to refactor, since most of the complexity is in `WatchStatus`, which does implement `ToXContentObject`.